test.js

```diff
@@ -249,6 +257,23 @@ const providerWithDot = /* tsx */ `
 export const Test = () => <Test.Provider />;
 `;
 
+const ternary = /* tsx */ `
+export const TernaryComponent = () => {
+  const active = useIsActive();
+  return active ? <ActiveComponent /> : null;
+};
+`;
+
+const ifBlock = /* tsx */ `
+export const IfBlockComponent = () => {
+  const active = useIsActive();
+  if (active) {
+    return <ActiveComponent />;
+  }
+  return <div />;
+};
+`;
```

Comment:
This test case is currently failing, I think it wants to add `data-component` to the returned `div` at the end.

Reply:
Yes this is similar to the null option we discussed. If we did handle this, we would need to evaluate all branches to see if there was another return somewhere else. And we can't just look through the tree for any return statement because it could be inside something like a callback or a nested function declaration which we would want to ignore. 

I'm not saying this isn't possible, but my gut says it would be very complicated to get right. I think, for now, we should either expect the data-component here or maybe get rid of this test.